boards/pic32-wifire/pic32_config_settings.c

```diff
@@ -0,0 +1,403 @@
+    0xffffffff /* unused bits must be 1 */
+    & (~_DEVCFG3_USERID_MASK     |  0xFFFF   << _DEVCFG3_USERID_POSITION)
+    & (~_DEVCFG3_FMIIEN_MASK     |  0        << _DEVCFG3_FMIIEN_POSITION)
+    & (~_DEVCFG3_FETHIO_MASK     |  1        << _DEVCFG3_FETHIO_POSITION)
+    & (~_DEVCFG3_PGL1WAY_MASK    |  0        << _DEVCFG3_PGL1WAY_POSITION)
+    & (~_DEVCFG3_PMDL1WAY_MASK   |  0        << _DEVCFG3_PMDL1WAY_POSITION)
+    & (~_DEVCFG3_IOL1WAY_MASK    |  0        << _DEVCFG3_IOL1WAY_POSITION)
+    & (~_DEVCFG3_FUSBIDIO_MASK   |  0        << _DEVCFG3_FUSBIDIO_POSITION);
+
+volatile uint32_t DEVCFG3_B1  __attribute__((used, section(".devcfg3_b1"))) =
+    0xffffffff /* unused bits must be 1 */
+    & (~_DEVCFG3_USERID_MASK     |  0xFFFF   << _DEVCFG3_USERID_POSITION)
+    & (~_DEVCFG3_FMIIEN_MASK     |  0        << _DEVCFG3_FMIIEN_POSITION)
+    & (~_DEVCFG3_FETHIO_MASK     |  1        << _DEVCFG3_FETHIO_POSITION)
+    & (~_DEVCFG3_PGL1WAY_MASK    |  0        << _DEVCFG3_PGL1WAY_POSITION)
+    & (~_DEVCFG3_PMDL1WAY_MASK   |  0        << _DEVCFG3_PMDL1WAY_POSITION)
+    & (~_DEVCFG3_IOL1WAY_MASK    |  0        << _DEVCFG3_IOL1WAY_POSITION)
+    & (~_DEVCFG3_FUSBIDIO_MASK   |  0        << _DEVCFG3_FUSBIDIO_POSITION);
+
+volatile uint32_t ADEVCFG3_B1 __attribute__((used, section(".adevcfg3_b1"))) =
+    0xffffffff /* unused bits must be 1 */
+    & (~_DEVCFG3_USERID_MASK     |  0xFFFF   << _DEVCFG3_USERID_POSITION)
+    & (~_DEVCFG3_FMIIEN_MASK     |  0        << _DEVCFG3_FMIIEN_POSITION)
+    & (~_DEVCFG3_FETHIO_MASK     |  1        << _DEVCFG3_FETHIO_POSITION)
+    & (~_DEVCFG3_PGL1WAY_MASK    |  0        << _DEVCFG3_PGL1WAY_POSITION)
+    & (~_DEVCFG3_PMDL1WAY_MASK   |  0        << _DEVCFG3_PMDL1WAY_POSITION)
+    & (~_DEVCFG3_IOL1WAY_MASK    |  0        << _DEVCFG3_IOL1WAY_POSITION)
+    & (~_DEVCFG3_FUSBIDIO_MASK   |  0        << _DEVCFG3_FUSBIDIO_POSITION);
+/* Not needed by default: */
+/* volatile uint32_t DEVCFG3_B2 __attribute__((used,section(".devcfg3_b2"))) = DEVCFG3_LA;   */
+/* volatile uint32_t ADEVCFG3_B2 __attribute__((used,section(".adevcfg3_la"))) = DEVCFG3_LA; */
+
+/*
+ * DEVCFG2_LA  @ 0x1FC0FFC4 (lower alias)
+ * ADEVFGC2_LA @ 0x1FC0FF44 (alternate devcfg2 in lower alias)
+ * DEVCFG2_B1  @ 0x1FC4FFC4 (config space 1)
+ * ADEVCFG2_B1 @ 0x1FC4FF44 (alternate devcfg2 in config space 1)
+ * DEVCFG2_B2  @ 0x1FC6FFC4 (config space 1)
+ * ADEVCFG2_B2 @ 0x1FC6FF44 (alternate devcfg2 in config space 2)
+ *
+ *  24MHz OSC / 3 * 50 / 2 = 200MHz
+ *
+ *  FPLLIDIV    DIV_3           System PLL Input Divider 3x Divider
+ *  FPLLRNG     RANGE_5_10_MHZ  System PLL Input Range   5-10 MHz Input
+ *  FPLLICLK    PLL_POSC        System PLL Input Clock Selection POSC is input to the System PLL
+ *  FPLLMULT    MUL_50          System PLL Multiplier    PLL Multiply by 50
+ *  FPLLODIV    DIV_2           System PLL Output Clock Divider    2x Divider
+ *  UPLLFSEL    FREQ_24MHZ      USB PLL Input Frequency Selection  USB PLL input is 24 MHz
```

Comment:
Potentially yes, but it would all have to be calculated by the pre-processor as these values need to be fixed in the final binary image (as mentioned above).